src/library/baseplaylistfeature.cpp

```diff
@@ -386,6 +383,47 @@ void BasePlaylistFeature::slotImportPlaylist() {
     }
 }
 
+void BasePlaylistFeature::slotCreateImportPlaylist() {
+    if (!m_pPlaylistTableModel) return;
+    
+    // Get file to read
+    QString playlist_file = getPlaylistFile();
+    if (playlist_file.isEmpty()) return;
+    
+    QFileInfo fileName(playlist_file);
+    m_pConfig->set(ConfigKey("[Library]","LastImportExportPlaylistDirectory"),
+                ConfigValue(fileName.dir().absolutePath()));
+    
+    // Get a valid name
+    QString baseName = fileName.baseName();
+    QString name;
+    bool validNameGiven = false;
+    int i = 0;
+    while (!validNameGiven) {
+        name = baseName;
+        if (i != 0) name += QString::number(i);
+        
+        // Check name
+        int existingId = m_playlistDao.getPlaylistIdFromName(name);
+        
+        validNameGiven = (existingId == -1);
+        ++i;
+    }
+    
+    int playlistId = m_playlistDao.createPlaylist(name);
+    
+    if (playlistId != -1) activatePlaylist(playlistId);
+    else {
```

Comment:
Style nit: If the else branch has braces, the if branch should have braces too. 
(I personally prefer to use always braces)

Below we have 4 space indent to much.